module/zfs/spa_stats.c

```diff
@@ -153,9 +152,7 @@ spa_read_history_init(spa_t *spa)
 	ssh->size = 0;
 	ssh->private = NULL;
 
-	(void) snprintf(name, KSTAT_STRLEN, "zfs/%s", spa_name(spa));
-
-	ksp = kstat_create(name, 0, "reads", "misc",
+	ksp = kstat_create("", 0, "reads", "misc",
```

Comment:
@loli10K @behlendorf..Thanks for the feedback. I agree with the points laid out here. 

kmem_asprintf() is definitely is better choice for dynamically allocating the string. Will update the patch soon.